server/typedb.rs

```diff
@@ -251,6 +260,24 @@ impl Server {
         }
         println!("Ready!");
     }
+
+    async fn shutdown_handler(shutdown_signal_sender: tokio::sync::watch::Sender<()>) {
+        Self::block_and_listen_ctrl_c().await;
+        println!("\nReceived CTRL-C. Initiating shutdown...");
+        shutdown_signal_sender.send(()).expect("Expected a successful shutdown signal");
+
+        tokio::spawn(Self::forced_shutdown_handler());
+    }
+
+    async fn forced_shutdown_handler() {
+        Self::block_and_listen_ctrl_c().await;
+        println!("\nReceived CTRL-C. Forcing shutdown...");
+        std::process::exit(1);
+    }
+
+    async fn block_and_listen_ctrl_c() {
+        tokio::signal::ctrl_c().await.expect("Failed to listen for CTRL-C signal");
+    }
```

Comment:
"Block" has a very specific meaning in `async`. This could just be called `ctrl_c()`, the use as `Self::ctrl_c().await` reads pretty well I think.
```suggestion
    async fn ctrl_c() {
        tokio::signal::ctrl_c().await.expect("Failed to listen for CTRL-C signal");
    }
```

Reply:
@flyingsilverfin `ctrl_c` or `listen_ctrl_c`? How readable is it for you?

Reply:
`await_ctlr_c` or `listen_ctlr_c`

Reply:
yeah ok i get dmitrii's argument, but i still find `listen_ctlr_c` easier to read without looking at the invocation.

Reply:
Done